### ISSUE TITLE
[Slider]: Not possible to know when it is invalid

### ISSUE THREAD

**Issue by TheSharpieOne** [carbon-design-system/carbon#9337]
## Environment

> Operating system: Any

> Browser: Any

## Detailed description

> What version of the Carbon Design System are you using?

> What did you expect to happen?

I expect to know when the field is invalid so I can trigger the `invalid` prop accordingly... or carbon components should already handle this. Why do I need to tell the component it is invalid when the component already knows?

> What happened instead?

When the user enters an invalid value, nothing gets called and there is no way to even know the user interacted with the field or that the field is invalid and thus no way to tell the component it is invalid (via the `invalid` prop) and no way to indicate it to the user.

> What WCAG 2.1 checkpoint does the issue violate?

## Steps to reproduce the issue

1. Go to https://react.carbondesignsystem.com/?path=/story/components-slider--playground
2. Open the actions to see events
3. Remove the value in the textbox (select it and backspace/delete it)
4. Notice No action is sent

## Additional information

This is regression from https://github.com/carbon-design-system/carbon/pull/8018

===== Accessibility checkpoint =====
Test Tool Type: IBM Equal Access Accessibility Checker
Checkpoint: IBM 3.3.1 Error Identification (https://www.ibm.com/able/requirements/requirements/#3_3_1)

=====STEPS TO REPRODUCE =====

Open Backup list page (https://cloud.ibm.com/cloud-storage/backup)
Login with ID & password
Open one backup instance
Go to Subscribers page
Click Add subscribers button
Clear the value of threshold
===== Problem Summary =====
It's better to let users know that an input error has occurred and can determine what is wrong. Users are offered suggestions to correct input errors. It is easier for people with visual impairment or cognitive disabilities to correct input errors.
![image](https://user-images.githubusercontent.com/664714/127178196-b19dee38-ad1c-482a-ac6c-bd4cda5e0934.png)

**Comment by tw15egan** [comment]
This makes sense, we'd welcome a PR to add in this functionality

**Comment by TheSharpieOne** [comment]
Fixed in #9132
